cdap-master/src/main/java/co/cask/cdap/master/environment/k8s/LogSaverServiceMain.java

```diff
@@ -0,0 +1,164 @@
+import co.cask.cdap.logging.appender.LogAppender;
+import co.cask.cdap.logging.appender.LogMessage;
+import co.cask.cdap.logging.framework.distributed.DistributedAppenderContext;
+import co.cask.cdap.logging.guice.LogQueryServerModule;
+import co.cask.cdap.logging.logbuffer.LogBufferService;
+import co.cask.cdap.logging.read.FileLogReader;
+import co.cask.cdap.logging.read.LogReader;
+import co.cask.cdap.logging.service.LogQueryService;
+import co.cask.cdap.logging.service.LogSaverStatusService;
+import co.cask.cdap.master.spi.environment.MasterEnvironment;
+import co.cask.cdap.master.spi.environment.MasterEnvironmentContext;
+import co.cask.cdap.messaging.guice.MessagingClientModule;
+import co.cask.cdap.proto.id.NamespaceId;
+import co.cask.cdap.security.auth.context.AuthenticationContextModules;
+import co.cask.cdap.security.authorization.AuthorizationEnforcementModule;
+import co.cask.cdap.security.impersonation.CurrentUGIProvider;
+import co.cask.cdap.security.impersonation.UGIProvider;
+import co.cask.http.HttpHandler;
+import com.google.common.util.concurrent.Service;
+import com.google.inject.AbstractModule;
+import com.google.inject.Injector;
+import com.google.inject.Module;
+import com.google.inject.PrivateModule;
+import com.google.inject.Scopes;
+import com.google.inject.multibindings.Multibinder;
+import com.google.inject.name.Names;
+import org.slf4j.ILoggerFactory;
+import org.slf4j.LoggerFactory;
+
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+import javax.annotation.Nullable;
+
+/**
+ * Main class for running log saver and log query service in Kubernetes.
+ */
+public class LogSaverServiceMain extends AbstractServiceMain {
+
+  /**
+   * Main entry point.
+   */
+  public static void main(String[] args) throws Exception {
+    main(LogSaverServiceMain.class, args);
+  }
+
+  @Override
+  protected List<Module> getServiceModules() {
+    return Arrays.asList(
+      new NamespaceQueryAdminModule(),
+      new AuthorizationEnforcementModule().getDistributedModules(),
+      new AuthenticationContextModules().getMasterModule(),
+      new MessagingClientModule(),
+      getDataFabricModule(),
+      // Always use local table implementations, which use LevelDB.
+      // In K8s, there won't be HBase and the cdap-site should be set to use SQL store for StructuredTable.
+      new SystemDatasetRuntimeModule().getStandaloneModules(),
+      new DataSetsModules().getStandaloneModules(),
+      new LocalLocationModule(),
+      // log handler is co-located with log saver
+      new LogQueryServerModule(),
+      new AbstractModule() {
+        @Override
+        protected void configure() {
+          // Current impersonation is not supported
+          bind(UGIProvider.class).to(CurrentUGIProvider.class).in(Scopes.SINGLETON);
+          bind(LogReader.class).to(FileLogReader.class);
+          bind(LogAppender.class).to(SysOutLogAppender.class).in(Scopes.SINGLETON);
```

Comment:
For now binding it to SysOutAppender as before. We are not binding it to `RemoteLogAppender` to avoid infinite loop.

Reply:
Combine this `AbstractModule` and the next `PrivateModule`.